src/Microsoft.NetCore.Analyzers/Core/Security/DoNotCallDangerousMethodsInDeserialization.cs

```diff
@@ -0,0 +1,243 @@
+
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var compilation = compilationStartAnalysisContext.Compilation;
+                    var serializableAttributeTypeSymbol = WellKnownTypes.SerializableAttribute(compilation);
+
+                    if (serializableAttributeTypeSymbol == null)
+                    {
+                        return;
+                    }
+
+                    var dangerousMethodSymbolsBuilder = ImmutableArray.CreateBuilder<IMethodSymbol>();
+                    var systemIOFileTypeSymbol = WellKnownTypes.SystemIOFile(compilation);
+
+                    if (systemIOFileTypeSymbol != null)
+                    {
+                        foreach (var targetMethodName in SystemIOFileMethodMetadataNames)
+                        {
+                            dangerousMethodSymbolsBuilder.AddRange(systemIOFileTypeSymbol.GetMembers(targetMethodName).Where(s => s is IMethodSymbol).Cast<IMethodSymbol>());
+                        }
+                    }
+
+                    var systemReflectionAssemblyTypeSymbol = WellKnownTypes.SystemReflectionAssembly(compilation);
+
+                    if (systemReflectionAssemblyTypeSymbol != null)
+                    {
+                        foreach (var targetMethodName in SystemReflectionAssemblyMethodMetadataNames)
+                        {
+                            dangerousMethodSymbolsBuilder.AddRange(systemReflectionAssemblyTypeSymbol.GetMembers(targetMethodName).Where(s => s is IMethodSymbol).Cast<IMethodSymbol>());
+                        }
+                    }
+
+                    var dangerousMethodSymbols = dangerousMethodSymbolsBuilder.ToImmutable();
+
+                    if (dangerousMethodSymbols.Length == 0)
+                    {
+                        return;
+                    }
+
+                    var attributeTypeSymbolsBuilder = ImmutableArray.CreateBuilder<INamedTypeSymbol>();
+                    var onDeserializingAttributeTypeSymbol = WellKnownTypes.OnDeserializingAttribute(compilation);
+
+                    if (onDeserializingAttributeTypeSymbol != null)
+                    {
+                        attributeTypeSymbolsBuilder.Add(onDeserializingAttributeTypeSymbol);
+                    }
+
+                    var onDeserializedAttributeTypeSymbol = WellKnownTypes.OnDeserializedAttribute(compilation);
+
+                    if (onDeserializedAttributeTypeSymbol != null)
+                    {
+                        attributeTypeSymbolsBuilder.Add(onDeserializedAttributeTypeSymbol);
+                    }
+
+                    var attributeTypeSymbols = attributeTypeSymbolsBuilder.ToImmutable();
+                    var streamingContextTypeSymbol = WellKnownTypes.StreamingContext(compilation);
+                    var IDeserializationCallback = WellKnownTypes.IDeserializationCallback(compilation);
```

Comment:
Nit: Name this variable `iDeserializationCallbackSymbol` (other variable names have "symbol" in them).